Release v1.2.0 (#45)
eProsima/Micro-XRCE-DDS-docs: docs/client_service.rst

On the other hand, the service is in charge of receiving the requests from the client, summing the two integers, and finally sending the response to the client.

Before considering the client and service examples, it is first necessary to briefly summarize how the `Requester` and `Replier` entities work,
as well as to list the functions related to both entities.

Requester
^^^^^^^^^

The `Requester` entity is composed of a `Publisher` and a `Subscriber` associated with a `RequestTopic` and a `ReplyTopic` respectively.
The `Publisher` is in charge of sending the request, while the `Susbscriber` receives the replies.

To create a `Requester` entity, the ``uxr_buffer_create_requester_xml`` or ``uxr_buffer_create_requester_ref`` shall be used.
Once created, requests can be sent through ``uxr_buffer_request``, and replies can be received, making a data request to the *Agent* using the ``uxr_buffer_request_data``.
Replies are received through the `on_reply` callback which shall be set by the ``uxr_set_reply_callback``.
This callback has a parameter `reply_id` which corresponds to the identifier returned by the ``uxr_buffer_request`` call.

Replier
^^^^^^^

> The `Reply` entity is a mirror of the `Requester`, that is, it contains a `Publisher` and a `Subscriber` also, but the topic association is reversed,

Reviewer: ```suggestion
The `Reply` entity is a mirror of the `Requester`, that is, it contains a `Publisher` and a `Subscriber` as well, but the topic association is reversed,
```

> the `Publisher` is associated with the `ReplyTopic` and the `Subscriber` to the `RequestTopic`.

Reviewer: ```suggestion
as the `Publisher` is associated with the `ReplyTopic` and the `Subscriber` to the `RequestTopic`.
```

> In that case, the `Subscriber` is in charge of receiving the request from the `Requester`, while the `Publisher` sends the replies.

Reviewer: ```suggestion
In this case, the `Subscriber` is in charge of receiving the request from the `Requester`, while the `Publisher` sends the replies.
```

            return 1;
        }

        // Request replies
        uxrDeliveryControl delivery_control = {0};
        delivery_control.max_samples = UXR_MAX_SAMPLES_UNLIMITED;
        uint16_t read_data_req = uxr_buffer_request_data(&session, reliable_out, requester_id, reliable_in, &delivery_control);

        // Write requests
        bool connected = true;
        uint32_t count = 0;
        while (connected)
        {
            uint8_t request[2 * 4] = {0};
            ucdrBuffer ub;

            ucdr_init_buffer(&ub, request, sizeof(request));
            ucdr_serialize_uint32_t(&ub, count);
            ucdr_serialize_uint32_t(&ub, count);

            uint16_t request_id = uxr_buffer_request(&session, reliable_out, requester_id, request, sizeof(request));
            printf("Request sent: (%d + %d) [id: %d]\n", count, count, request_id);
            connected = uxr_run_session_time(&session, 1000);

            ++count;
        }

        return 0;
    }

Service example
^^^^^^^^^^^^^^^

The code below shows the *ReplyAdder* example available at `eProsima Micro XRCE-DDS Client <https://github.com/eProsima/Micro-XRCE-DDS-Client/tree/master/examples/ReplyAdder>`__.

> This example represents a service application that receives requests composed by two integers, sums both number, and finally sends the response.

Reviewer: ```suggestion
This example represents a service application that receives requests composed by two integers, sums both numbers, and finally sends the response.
```
